lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,310 @@
+      sample: '0.0.0.0'
+    name:
+      description: The name of the process.
+      type: string
+      sample: 'rsyslogd'
+    pid:
+      description: The pid of the process.
+      type: int
+      sample: 609
+    port:
+      description: The port the pid was listening on.
+      type: int
+      sample: 514
+    protocol:
+      description: The protocol, TCP or UDP, of the listening connection.
+      type: string
+      sample: 'udp'
+    stime:
+      description: The start time of the process.
+      type: string
+      sample: 'Thu Feb  2 13:29:45 2017'
+    user:
+      description: The user running the process.
+      type: string
+      sample: 'root'
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'committer',
+                    'version': '1.0'}
+
+import re
+import platform
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+def netStatParse(raw):
+    results = list()
+    for line in iter(raw.splitlines()):
+        listening_search = re.search('[^ ]+:[0-9]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('([^ ]+):([0-9]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                protocol = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                protocol = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('(([0-9]+)/(.*)|-)', pidstr)
+            if conns and pids:
+                address = conns.group(1)
+                port = conns.group(2)
+                if (pids.group(2)):
+                    pid = pids.group(2)
+                else:
+                    pid = 0
+                if (pids.group(3)):
+                    name = pids.group(3)
+                else:
+                    name = ''
+                result = dict(pid=int(pid), address=address, port=int(port), protocol=protocol, name=name)
+                if result not in results:
+                    results.append(result)
+            else:
+                raise EnvironmentError('Could not get process information for the listening ports.')
+    return results
+
+def applyWhitelist(portspids, whitelist=list()):
+    processes = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            process = dict(pid=p['pid'], address=p['address'], port=p['port'], protocol=p['protocol'], name=p['name'], stime=p['stime'], user=p['user'])
+            if process not in processes:
+                processes.append(process)
+    return processes
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            whitelist_tcp = dict(required=False, type='list', default=list()),
+            whitelist_udp = dict(required=False, type='list', default=list())
+        ),
+        supports_check_mode=False
```

Comment:
For facts modules, it usually makes sense to set `support_check_mode=True` as the facts module is `safe` to run in check mode or in other words, if not set to `True` this module will be skipped in check mode.